lib/smooth/include/smooth/core/network/Wifi.h

```diff
@@ -73,6 +73,8 @@ namespace smooth::core::network
                                             void* event_data);
 
             [[nodiscard]] std::string get_mac_address() const;
+            esp_err_t get_local_mac_address(uint8_t m[6]) const;
+            ip4_addr_t get_local_ip() const;
```

Comment:
Just a thought, smooth has an IP4-class, could it be used as a return value?

Reply:
I only need the ip address as `uint32_t` or `uint8_t[4]`. The IP4 class additionally includes port number and other stuff which I don't need and which in the case of UDP and/or in the Wifi class does not make so much sense. In case of a TCP/IP socket connection these information makes more sense - so there could be a member function of the Socket class for getting such information?

Reply:
Fair point.

Btw, I think these two methods should be static; you're unlikely to have access to the Wifi-instance where you need them and I don't want to have to pass it around just to get the IP/mac.

Reply:
Isn't this valid for `std::string get_mac_address() const;` also?